enterprise/migrations/0122_remove_sync_enterprise_catalog_query_bool.py

```diff
@@ -0,0 +1,21 @@
+# Generated by Django 2.2.17 on 2021-01-29 19:25
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('enterprise', '0121_systemwiderole_add_ent_cust_field'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='enterprisecustomercatalog',
```

Comment:
it would be better to introduce this migration in a separate pr: this initial pr should focus on removing references to the field. see "How to Drop a column section" here: https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations

Reply:
I'm not sure it'll be that simple - check the test results from my initial commit which didn't include the migration https://github.com/edx/edx-enterprise/runs/1787640214 (`NOT NULL constraint failed`). Is there some sort of best-of-both-worlds scenario here?

Reply:
Given those failures it seems there will need to be 3 PRs/deployments:

1. Remove NOT NULL constraint on field
2. Remove all references to field
3. Perform migration

Lmk if there's anything I'm missing

Reply:
@chavesj On your initial commit, I believe you were seeing `NOT NULL constraint failed` error because the `sync_enterprise_catalog_query` field was removed from the model itself at the same time as removing its references. If this PR keeps that field in tact for, we should be able to just remove the references to it first.

If your initial commit did not remove the `sync_enterprise_catalog_query` field from the model and only removed its references throughout the code first, that should work. Doubling back to remove the field itself alongside creating the migration would then be done as a separate PR.